components/core/DataView.js

```diff
@@ -247,16 +263,95 @@ export default class DataView extends React.Component {
 
   _handleCheckScroll = Window.debounce(this._handleScroll, 200);
 
-  _handleCheckBox = (e) => {
+  _handleTableCheckBox = (e) => {
     let checked = this.state.checked;
+
+    let i = e.target.name;
+    if (this.state.isShiftDown && this.state.lastSelectedItemIndex !== i) {
+      return this._handleShiftClick({
+        currentSelectedItemIndex: i,
+        lastSelectedItemIndex: this.state.lastSelectedItemIndex,
+        checked,
+      });
+    }
+
     if (e.target.value === false) {
       delete checked[e.target.name];
-      this.setState({ checked });
+      this.setState({ checked, lastSelectedItemIndex: i });
       return;
     }
+
     this.setState({
       checked: { ...this.state.checked, [e.target.name]: true },
+      lastSelectedItemIndex: i,
     });
+
+    if (checked[i]) {
+      delete checked[i];
+    } else {
+      checked[i] = true;
+    }
+    this.setState({ checked, lastSelectedItemIndex: i });
+  };
+
+  /* (NOTE):daniel This disable text selection while pressing shift key */
+  _handleSelectStart = (e) => {
+    if (this.state.isShiftDown) {
+      e.preventDefault();
+    }
+  };
+
+  _handleKeyUp = (e) => {
+    if (e.shiftKey && this.state.isShiftDown) {
+      this.setState({ isShiftDown: false });
+    }
+  };
+
+  _handleKeyDown = (e) => {
+    if (e.shiftKey && !this.state.isShiftDown) {
+      this.setState({ isShiftDown: true });
+    }
+  };
+
+  _handleGridCheckBox = (e, i) => {
```

Comment:
I don't think you need a separate _handleGridCheckBox and _handleTableCheckBox. These two do the same thing (though there's a small error in _handleTableCheckBox, so I'd keep just _handleGridCheckBox

Reply:
You just have to pass in the index to it the same way that is done on line 597, since handleGridCheckBox uses a passed in value `i` rather than the name attached to the thing that fired the event